src/rlp/tests.rs

```diff
@@ -81,6 +85,106 @@ mod rlc {
         MockProver::run(k, &circuit, vec![]).unwrap().assert_satisfied();
     }
 
+    #[test_case([0,0,0,0].map(Fr::from).to_vec(); "RLC([0,0,0,0]) = 0")]
+    pub fn test_rlc_chip_zero<F: ScalarField>(inputs: Vec<F>) {
+        let mut builder = RlcThreadBuilder::mock();
+        let ctx = builder.gate_builder.main(0);
+        let inputs = ctx.assign_witnesses(inputs);
+        let len = ctx.load_witness(F::from(inputs.len() as u64));
+
+        let circuit = RlcCircuitBuilder::new(
+            builder,
+            None,
+            move |builder: &mut RlcThreadBuilder<F>, rlc: &RlcChip<F>| {
+                let gate = GateChip::default();
+                let (ctx_gate, ctx_rlc) = builder.rlc_ctx_pair();
+                let rlc_trace = rlc.compute_rlc((ctx_gate, ctx_rlc), &gate, inputs, len);
+                let rlc_val = *rlc_trace.rlc_val.value();
+                assert_eq!(rlc_val, F::from(0));
+            },
+        );
+
+        circuit.config(DEGREE as usize, Some(6));
+        MockProver::run(DEGREE, &circuit, vec![]).unwrap().assert_satisfied();
+    }
+
+    #[test_case(([1, 2, 3, 4, 5, 6, 7, 8].map(Fr::from).to_vec(), [1, 2, 3, 4, 5, 6, 7, 8].map(Fr::from).to_vec()) ; "Dynamic RLC test, var len 1")]
+    #[test_case(([1, 2, 3].map(Fr::from).to_vec(), [1, 2, 3, 4, 5, 6, 7, 8].map(Fr::from).to_vec()) ; "Dynamic RLC test, var len 2")]
+    pub fn test_rlc_dynamic_var_len<F: ScalarField>(inputs: (Vec<F>, Vec<F>)) {
+        let mut rlc_builder = RlcThreadBuilder::mock();
+        let mut builder = GateThreadBuilder::mock();
+        let ctx = builder.main(0);
+
+        let a_b = ctx.assign_witnesses(inputs.0.iter().chain(inputs.1.iter()).cloned());
+        let combined_len = inputs.0.len() as u64 + inputs.1.len() as u64;
+        let a_b_len = ctx.load_witness(F::from(combined_len));
+
+        let a_len = ctx.load_witness(F::from(inputs.0.len() as u64));
+        let mut a = inputs.0.clone();
+        a.resize(combined_len as usize, F::from(0));
+        let a = ctx.assign_witnesses(a);
+
+        let b_len = ctx.load_witness(F::from(inputs.1.len() as u64));
+        let mut b = inputs.1.clone();
+        b.resize(combined_len as usize, F::from(0));
+        let b = ctx.assign_witnesses(b);
+
+        let gate = GateChip::default();
+        let (ctx_gate, ctx_rlc) = rlc_builder.rlc_ctx_pair();
+        let rlc = RlcChip::new(F::from(2));
+        let rlc_trace = rlc.compute_rlc((ctx_gate, ctx_rlc), &gate, a_b, a_b_len);
+        let rlc_val = *rlc_trace.rlc_val.value();
+
+        let rlc_trace_a = rlc.compute_rlc((ctx_gate, ctx_rlc), &gate, a, a_len);
```

Comment:
Should also test inputs where `a_len` is not equal to `a.len()`.  This probably requires adding additional test case inputs for `a_len`, `b_len`, and `a_b_len`.

Reply:
Will also be interesting to add edge cases for `a_len = 0` or `b_len = 0`.

Reply:
Would be nice to have some test cases where `a_len < a.len()` and `b_len < b.len()` but `a_len + b_len = a_b_len`.  OK to merge after adding.